packages/cli/src/utils.ts

```diff
@@ -453,3 +454,15 @@ function sortOas3Keys(document: Oas3Definition): Oas3Definition {
   // merge any other top-level keys (e.g. vendor extensions)
   return Object.assign(result, document);
 }
+
+export function checkIfRulesetExist(rules: typeof StyleguideConfig.prototype.rules) {
+  const ruleset = {
+    ...rules.oas2,
+    ...rules.oas3_0,
+    ...rules.oas3_0,
+  };
+
+  if (isEmptyObject(ruleset)) {
+    exitWithError('No rules were configured. Please configure the rules in the configuration file');
```

Comment:
Should we exit with the error? I thought it had to be a warning.

Reply:
I think the "no rules" case could be a warning, since we do allow that configuration, but we should avoid showing the message with the :tada:  in it saying things are valid if we didn't actually validate anything.

Reply:
We say this currently:

> Woohoo! Your OpenAPI definitions are valid. 🎉

We should say this maybe:

> ⚠️ No rules were configured. Learn how to configure rules: https://redocly.com/docs/cli/rules/

Reply:
I think it should exit with an error only for the particular case when someone runs the lint command. If you are running the lint command and relying on the exit code, you should be aware that lint didn't actually run when you have no rules configured.

Reply:
I think the warning is more appropriate for some contexts, and we do allow a no-rules configuration as valid. The behaviour should be identical whether linting is happening with the lint command or as a side effect to bundle - and turning off rules to get a bundle of whatever the current state is could be a valid use case IMO.

Reply:
In my opinion, the fact that we have currently-passing pipelines is a big reason to stay with a warning. Breaking someone's pipelines makes work for our users and puts them off upgrading in future. I also don't think that no rules is always an error case - that's for the user to decide how to use the tools we give them; outputting the warning gives them the information they need.

Reply:
I think this is the problem:

> In my opinion, the fact that we have currently-passing pipelines is a big reason to stay with a warning. Breaking someone's pipelines makes work for our users and puts them off upgrading in future. 

This argument doesn't work for me. Let's say a popular unit testing tool had a bug where they didn't report any errors with code. This allowed developers to ship broken code. Now, shouldn't they fix the issue so that tests resume working? (Which means pipelines will fail.) 

If they updated to Redocly CLI  1.0.0-beta.113 to 1.0.0-beta.125, then they are running lint without actually linting anything. This leads them to ship something invalid in production, which is much more devastating than a pipeline not passing. You don't need to lint to bundle the definition. If they were new to Redocly, they may think that it is doing something when it isn't doing anything. If they were pre- 1.0.0-beta.113 users who upgraded, then it used to do something and it doesn't do that any more.

If they are prior to 1.0.0-beta.113, then to keep a similar behavior we should show the error. I consider that we didn't throw an error a bug, and we are potentially causing damage to people who think they are linting when they are not linting.

Reply:
I agree that we should exit with the error. I can't remember a time when I was looking through success pipeline logs to see if there are any warnings 😅.

The tool is still in beta so I don't think this is too bad to break pipelines at this stage. I also believe we don't have too many users impacted by that.

Alternatively, we can keep the warning for now and delay the breaking change until GA. But I lean towards changing it now.

Reply:
Thanks for chiming in @RomanHotsiy . Let's implement it with errors.